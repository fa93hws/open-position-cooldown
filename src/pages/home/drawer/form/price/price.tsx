import * as React from 'react';
import { Typography, Box } from '@material-ui/core';

import type { InputStore } from '../input/input-store';
import { createInput, ExposedInputProps } from '../input/input';
import { nonEmpty } from '../validator';

type PriceProps = {
  InputComponent: React.ComponentType<ExposedInputProps>;
};
const Price = React.memo((props: PriceProps) => (
  <Box>
    <Typography variant="h6" component="h4" gutterBottom>
      为什么当前价格可以买入？
    </Typography>
    <props.InputComponent multiline placeholder="可换行" />
  </Box>
));

export function createPriceExplain(): [React.ComponentType, InputStore] {
  const [InputComponent, store] = createInput([nonEmpty]);
  const Component = () => <Price InputComponent={InputComponent} />;
  return [Component, store];
}
