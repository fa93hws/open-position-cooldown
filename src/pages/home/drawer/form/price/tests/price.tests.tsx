import { cleanup } from '@testing-library/react';
import * as React from 'react';

import { createPriceExplain } from '../price';

describe('Price', () => {
  beforeEach(cleanup);

  it('renders the following', () => {
    const [Component] = createPriceExplain();
    expect(<Component />).toMatchRenderedSnapshot();
  });
});
