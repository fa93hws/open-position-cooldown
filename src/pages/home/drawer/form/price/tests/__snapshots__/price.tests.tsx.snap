// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price renders the following 1`] = `
<div
  class="MuiBox-root MuiBox-root-1"
>
  <h4
    class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
  >
    为什么当前价格可以买入？
  </h4>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline MuiInputBase-marginDense MuiOutlinedInput-marginDense"
    >
      <textarea
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
        placeholder="可换行"
        rows="1"
        style="height: 0px; overflow: hidden;"
      />
      <textarea
        aria-hidden="true"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
        readonly=""
        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
        tabindex="-1"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-3"
          style="width: 0.01px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
